python/cudf/cudf/_fuzz_testing/utils.py

```diff
@@ -192,11 +222,79 @@ def pandas_to_avro(df, file_name=None, file_io_obj=None):
         fastavro.writer(file_io_obj, avro_schema, records)
 
 
+def _preprocess_to_orc_tuple(df):
+    def _null_to_None(value):
+        if value is pd.NA or value is pd.NaT:
+            return None
+        else:
+            return value
+
+    has_nulls_or_nullable_dtype = any(
+        [
+            True
+            if df[col].dtype in pandas_dtypes_to_cudf_dtypes
+            or df[col].isnull().any()
+            else False
+            for col in df.columns
+        ]
+    )
+
+    tuple_list = []
+    for tup in df.itertuples(index=False, name=None):
+        if has_nulls_or_nullable_dtype:
+            tuple_list.append(tuple(map(_null_to_None, tup)))
+        else:
+            tuple_list.append(tup)
```

Comment:
Can use a list comprehension here:
```suggestion
    tuple_list = [tuple(map(_null_to_None, tup)) if has_nulls_or_nullable_dtype else tup for tup in df.itertuples(index=False, name=None)]
```